### ISSUE TITLE
Decryption key format

### ISSUE THREAD

**Issue by valki2** [DomiStyle/esphome-dlms-meter#34]
Hi!
For those who get problems when decrypting:

when your key is 32 digits like D658345DF620A2839E5AD6A6D1B85135

then you have to fill each pair of the key into the 0x00 format... like:

0xD6, 0x58, 0x34, 0x5D, 0xF6, 0x20, 0xA2, 0x83, 0x9E, 0x5A, 0xD6, 0xA6, 0xD1, 0xB8, 0x51, 0x35

Have fun!

**Comment by valki2** [comment]
ok looks like the data is truncated as there is no "16" (Mbus Stop) at the end.

**Comment by ert666** [comment]
@valki2 
So your code would be...
...
      uint8_t key[] = {0xD6, 0x58, 0x34, 0x5D, 0xF6, 0x20, 0xA2, 0x83, 0x9E, 0x5A, 0xD6, 0xA6, 0xD1, 0xB8, 0x51, 0x35};
      dlms_meter->set_key(key, 32; // Pass your decryption key and key length here
...
?

**Comment by kenci** [comment]
@ert666 no, it should be `dlms_meter->set_key(key, 16);` because there are 16 bytes.
